### PR TITLE
 Added BinBuilder class

### PR DESCRIPTION
**Pull request by FarzanRashid**
This PR contains the BinBuilder class for roulette game.

### CODE REVIEW

File: src/roulette.py [hunk 1]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+
+    @staticmethod
+    def create_straight_bets(wheel: Wheel) -> None:
+        odd_for_straight_bet = 35
+        straight_bet_numbers = set(_ for _ in range(37))
+
+        for i in straight_bet_numbers:
+            straight_bet = Outcome(str(i), odd_for_straight_bet)
+            wheel.addOutcome(i, straight_bet)
+        zero_zero_bet = Outcome("00", 35)
```
Comment (FarsanRashid) [inline src/roulette.py:146]: can 35 be replaced with `odd_for_straight_bet`?

File: src/roulette.py [hunk 2]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+class BinBuilder:
+    def __init__(self) -> None:
+        pass
+
+    def buildBins(self, wheel: Wheel) -> None:
+        self.create_straight_bets(wheel)
+
+    @staticmethod
+    def create_straight_bets(wheel: Wheel) -> None:
+        odd_for_straight_bet = 35
```
Comment (FarsanRashid) [inline src/roulette.py:140]: can rename with `straight_bet_odd`? (applies for other bet types)

File: src/roulette.py [hunk 3]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+        bin_index_of_zero_zero_outcome = 37
+        wheel.addOutcome(bin_index_of_zero_zero_outcome, zero_zero_bet)
+
+    @staticmethod
+    def create_horizontal_split_bets(wheel: Wheel) -> None:
+        odds_for_split_bet = 17
+
+        first_column_numbers = set()
+        for i in range(12):
+            first_column_numbers.add(i * 3 + 1)
```
Comment (FarsanRashid) [inline src/roulette.py:156]: Should be done using comprehensions. Can check item _6.2 Comprehending Comprehensions_ of Python Tricks book.

File: src/roulette.py [hunk 4]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+        second_column_numbers = set()
+        for i in range(12):
+            second_column_numbers.add(i * 3 + 2)
+        for i in second_column_numbers:
+            column_two_three_split_bet = Outcome(str(i) + "-" + str(i + 1), odds_for_split_bet)
+            bin_indexes_for_horizontal_split_bets = i, i + 1
+            wheel.addOutcome(bin_indexes_for_horizontal_split_bets[0],
+                             column_two_three_split_bet)
+            wheel.addOutcome(bin_indexes_for_horizontal_split_bets[1],
+                             column_two_three_split_bet)
```
Comment (FarsanRashid) [inline src/roulette.py:172]: The two blocks for different column numbers, does the same thing. Only difference is the for loops iterate over different set of column numbers. To achieve DRY, Better have a single set which have both column numbers and iterate over them once. The set can be built as 
`result_set = {i * 3 + 2 for i in range(12)} | {i * 3 + 2 for i in range(12)}`

File: src/roulette.py [hunk 5]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+
+    @staticmethod
+    def create_vertical_split_bets(wheel: Wheel) -> None:
+        numbers_to_create_vertical_split_bets = set(_ for _ in range(1, 34))
```
Comment (FarsanRashid) [inline src/roulette.py:176]: better rename to `vertical_split_bet_numbers`?

File: src/roulette.py [hunk 6]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+        for i in numbers_to_create_vertical_split_bets:
```
Comment (FarsanRashid) [inline src/roulette.py:177]: better rename `i ` to `number`

File: src/roulette.py [hunk 7]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+            vertical_split_bet = Outcome(str(i) + "-" + str(i + 3), 17)
+            bin_indexes_for_vertical_split_outcome = i, i + 3
+            wheel.addOutcome(bin_indexes_for_vertical_split_outcome[0], vertical_split_bet)
+            wheel.addOutcome(bin_indexes_for_vertical_split_outcome[1], vertical_split_bet)
+
```
Comment (FarsanRashid) [inline src/roulette.py:182]: no need to create `bin_indexes_for_vertical_split_outcome`, calling `wheel.addOutcome(..)` with `i` and `i+3` should be enough.

File: src/roulette.py [hunk 8]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+    @staticmethod
+    def create_street_bets(wheel: Wheel) -> None:
+        odds_for_street_bet = 11
+        numbers_to_create_street_bets = set()
+        for i in range(12):
+            numbers_to_create_street_bets.add(i * 3 + 1)
```
Comment (FarsanRashid) [inline src/roulette.py:188]: can be simplified using comprehension

File: src/roulette.py [hunk 9]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+        for i in numbers_to_create_street_bets:
+            street_bet = Outcome(str(i) + "-" + str(i + 1) + "-" + str(i + 2), odds_for_street_bet)
+            indexes_for_street_bets = i, i + 1, i + 2
+            wheel.addOutcome(indexes_for_street_bets[0], street_bet)
+            wheel.addOutcome(indexes_for_street_bets[1], street_bet)
+            wheel.addOutcome(indexes_for_street_bets[2], street_bet)
```
Comment (FarsanRashid) [inline src/roulette.py:194]: can be called directly using `i, i + 1, i +2`

File: src/roulette.py [hunk 10]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+
+    @staticmethod
+    def create_corner_bets(wheel: Wheel) -> None:
+        odds_for_corner_bet = 8
+
+        first_column_numbers = set()
+        for i in range(11):
+            first_column_numbers.add(i * 3 + 1)
+        for i in first_column_numbers:
+            first_column_corner_bet = Outcome(str(i) + "-" + str(i + 1) + "=" + str(i + 3) +
```
Comment (FarsanRashid) [inline src/roulette.py:204]: is `=` typo? should there be `-` between `(i + 3) and (i + 4)`

File: src/roulette.py [hunk 11]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+            wheel.addOutcome(indexes_for_line_bets[1], line_bet)
+            wheel.addOutcome(indexes_for_line_bets[2], line_bet)
+            wheel.addOutcome(indexes_for_line_bets[3], line_bet)
+            wheel.addOutcome(indexes_for_line_bets[4], line_bet)
+            wheel.addOutcome(indexes_for_line_bets[5], line_bet)
+
+    @staticmethod
+    def generate_five_bets(wheel: Wheel) -> None:
+        odds_for_five_bet = 6
+        five_bet = Outcome("00-0-1-2-3", odds_for_five_bet)
```
Comment (FarsanRashid) [inline src/roulette.py:245]: better create `five_bet_name` variable to avoid magic string

File: src/roulette.py [hunk 12]
```diff
@@ -126,3 +126,124 @@ def get(self, bin: int) -> Bin:
...
+        bin_indexes_for_five_bet = (0, 1, 2, 3, 37)
+
+        for i in bin_indexes_for_five_bet:
```
Comment (FarsanRashid) [inline src/roulette.py:248]: better replace `i` with `index`

File: src/roulette.py [hunk 13]
```diff
@@ -281,3 +281,20 @@ def create_even_money_bets(wheel: Wheel) -> None:
                 wheel.addOutcome(number, red_bet)
             if number not in red_bet_numbers:
                 wheel.addOutcome(number, black_bet)
+
+    @staticmethod
+    def create_dozen_bets(wheel: Wheel) -> None:
+        dozen_bet_odds = 2
+        for number in range(3):
```
Comment (FarsanRashid) [inline src/roulette.py:288]: rename `number` to `dozen`

File: src/roulette.py [hunk 14]
```diff
@@ -134,119 +134,99 @@ def __init__(self) -> None:
...
-            wheel.addOutcome(bin_indexes_for_horizontal_split_bets[0],
-                             column_two_three_split_bet)
-            wheel.addOutcome(bin_indexes_for_horizontal_split_bets[1],
-                             column_two_three_split_bet)
+        split_bet_odds = 17
+
+        split_bet_numbers = {number * 3 + 1 for number in range(12)} | {number * 3 + 2 for number
+                                                                        in range(12)}
+        for number in split_bet_numbers:
+            column_one_two_split_bet = Outcome(str(number) + "-" + str(number + 1),
```
Comment (FarsanRashid) [inline src/roulette.py:166]: `column_one_two_split_bet` is no longer meaningful as `first_column_numbers` has been removed.

File: src/roulette.py [hunk 15]
```diff
@@ -126,3 +126,172 @@ def get(self, bin: int) -> Bin:
...
+        self.create_straight_bets(wheel)
+        self.create_horizontal_split_bets(wheel)
+        self.create_vertical_split_bets(wheel)
+        self.create_street_bets(wheel)
+        self.create_corner_bets(wheel)
+        self.create_line_bets(wheel)
+        self.generate_five_bets(wheel)
+        self.create_even_money_bets(wheel)
+        self.create_dozen_bets(wheel)
+        self.create_column_bets(wheel)
```
Comment (FarsanRashid) [inline src/roulette.py:145]: `buildBins` is responsible to invoke helper methods that build/populate bins of wheel. The helper methods are not creating bets (in true sense) so method names like `create_...` does not seem appropriate. Instead `build_bins_for_x_bet` seems better choice as the name reflects the purpose.

File: src/roulette.py [hunk 16]
```diff
@@ -126,3 +126,172 @@ def get(self, bin: int) -> Bin:
...
+
+    @staticmethod
+    def create_straight_bets(wheel: Wheel) -> None:
+        straight_bet_odds = 35
+        straight_bet_numbers = {_ for _ in range(37)}
```
Comment (FarsanRashid) [inline src/roulette.py:150]: pylint reports "Unnecessary use of a comprehension, use set(range(37)) instead."

File: src/roulette.py [hunk 17]
```diff
@@ -126,3 +126,172 @@ def get(self, bin: int) -> Bin:
...
+
+        for number in straight_bet_numbers:
+            straight_bet = Outcome(str(number), straight_bet_odds)
```
Comment (FarsanRashid) [inline src/roulette.py:153]: `straight_bet` is an `Outcome` object, not some bet class object. Better rename to `straight_bet_outcome` or simply `outcome`.

File: src/roulette.py [hunk 18]
```diff
@@ -126,3 +126,172 @@ def get(self, bin: int) -> Bin:
...
+        five_bet = Outcome(five_bet_name, odds_for_five_bet)
+        bin_indexes_for_five_bet = {0, 1, 2, 3, 37}
+
+        for index in bin_indexes_for_five_bet:
+            wheel.addOutcome(index, five_bet)
+
+    @staticmethod
+    def create_even_money_bets(wheel: Wheel) -> None:
+        even_bet_odds = 1
+        red_bet_name = "Red"
```
Comment (FarsanRashid) [inline src/roulette.py:234]: a blank line odds and bet names will improve readability.

File: src/roulette.py [hunk 19]
```diff
@@ -126,3 +126,172 @@ def get(self, bin: int) -> Bin:
...
+
+        even_bet_numbers = {_ for _ in range(1, 37)}
+        red_bet_numbers = {1, 3, 5, 7, 9, 12, 14, 16, 18, 19, 21, 23, 25, 27, 30, 32, 34, 36}
+
+        red_bet = Outcome(red_bet_name, even_bet_odds)
+        black_bet = Outcome(black_bet_name, even_bet_odds)
+        low_bet = Outcome(low_bet_name, even_bet_odds)
+        high_bet = Outcome(high_bet_name, even_bet_odds)
+        even_bet = Outcome(even_bet_name, even_bet_odds)
+        odd_bet = Outcome(odd_bet_name, even_bet_odds)
```
Comment (FarsanRashid) [inline src/roulette.py:249]: pylint complains about "Too many local variables (17/15)".  what about creating Outcome in place while invoking `addOutcome`

File: tests/test_bin_builder.py [hunk 20]
```diff
@@ -0,0 +1,22 @@
+from unittest import TestCase
+from roulette import Outcome, Wheel, BinBuilder
+
+
+class TestBinBuilder(TestCase):
+    def setUp(self):
+        self.bin_builder = BinBuilder()
+        self.wheel = Wheel()
+
+    def test_if_straight_outcomes_are_generated_properly(self):
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:10]: better remove the `if` from test name

Comment (FarsanRashid) [inline tests/test_bin_builder.py:10]: it should also be tested that other bins except expected bins are empty.

Comment (FarzanRashid) [reply tests/test_bin_builder.py]: Done. Thanks.

File: src/roulette.py (hunk 10 above)
Comment (FarzanRashid) [reply src/roulette.py]: Fixed. Thanks.

File: src/roulette.py (hunk 12 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 11 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 13 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 9 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 8 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 7 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 6 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 5 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 4 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 1 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 2 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 3 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.Done. Thanks.

File: src/roulette.py (hunk 14 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 15 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 16 above)
Comment (FarzanRashid) [reply src/roulette.py]: Fixed. Thanks.

File: src/roulette.py (hunk 17 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done. Thanks.

File: src/roulette.py (hunk 18 above)
Comment (FarzanRashid) [reply src/roulette.py:398]: Done. Thanks.

File: src/roulette.py [hunk 21]
```diff
@@ -126,3 +126,381 @@ def get(self, bin: int) -> Bin:
...
+    """
+    :class:`BinBuilder` creates the :class:`Outcome` instances for all of the 38 individual
+    :class:`Bin` on a Roulette wheel.
+    """
+
+    def __init__(self) -> None:
+        """
+        Initializes the :class:`BinBuilder`.
+        """
+        pass
```
Comment (FarsanRashid) [inline src/roulette.py:141]: pylint reports unnecessary pass.

Comment (FarzanRashid) [reply src/roulette.py]: Done.

File: tests/test_bin_builder.py [hunk 22]
```diff
@@ -152,3 +153,26 @@ def test_bins_are_filled_for_even_money_bets(self):
...
+            "roulette.BinBuilder.build_bins_for_dozen_bets",
+            "roulette.BinBuilder.build_bins_for_column_bets",
+
+        }
+
+        mock_helper_method = Mock(name="mock_methods", return_value=Mock())
+
+        for method in helper_methods:
+            with patch(method, new=mock_helper_method):
+                self.bin_builder.buildBins(self.wheel)
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:176]: `buildBins` should be invoked only once

File: tests/test_bin_builder.py [hunk 23]
```diff
@@ -152,3 +153,26 @@ def test_bins_are_filled_for_even_money_bets(self):
...
+                mock_helper_method.assert_called()
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:177]: Redundant assert. `assert_called_with` is sufficient.

File: tests/test_bin_builder.py [hunk 24]
```diff
@@ -152,3 +153,26 @@ def test_bins_are_filled_for_even_money_bets(self):
                 self.assertIn(even_bet_outcome, self.wheel.bins[bin_index])
             if bin_index % 2 != 0:
                 self.assertIn(odd_bet_outcome, self.wheel.bins[bin_index])
+
+    def test_buildBins_invokes_other_helper_methods(self):
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:157]: Remove other from test name

File: tests/test_bin_builder.py [hunk 25]
```diff
@@ -0,0 +1,178 @@
...
+            "roulette.BinBuilder.build_bins_for_corner_bets",
+            "roulette.BinBuilder.build_bins_for_line_bets",
+            "roulette.BinBuilder.build_bins_for_five_bet",
+            "roulette.BinBuilder.build_bins_for_even_money_bets",
+            "roulette.BinBuilder.build_bins_for_dozen_bets",
+            "roulette.BinBuilder.build_bins_for_column_bets",
+
+        }
+
+        mock_helper_method = Mock(name="mock_methods", return_value=Mock())
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:172]: is it needed to set `return_value`?

File: tests/test_bin_builder.py [hunk 26]
```diff
@@ -0,0 +1,178 @@
...
+        self.wheel = Wheel()
+
+    def test_bins_are_filled_for_straight_bets(self):
+        straight_bet_odds = 35
+        straight_outcomes = {
+            0: Outcome("0", straight_bet_odds),
+            1: Outcome("1", straight_bet_odds),
+            36: Outcome("36", straight_bet_odds),
+            37: Outcome("00", straight_bet_odds)
+            }
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:18]: Instead of always testing against fixed bin ids, a better test strategy is to test against randomly generated bins along with special bins (e.g 0, 37)

File: src/roulette.py [hunk 27]
```diff
@@ -126,3 +126,380 @@ def get(self, bin: int) -> Bin:
...
+        """
+        split_bet_odds = 17
+
+        split_bet_numbers = set(range(1, 36))
+        for number in split_bet_numbers:
+            horizontal_split_bet_outcome = Outcome(
+                str(number) + "-" + str(number + 1), split_bet_odds
+            )
+            wheel.addOutcome(number, horizontal_split_bet_outcome)
+            wheel.addOutcome(number + 1, horizontal_split_bet_outcome)
```
Comment (FarsanRashid) [inline src/roulette.py:215]: horizontal/left right bet should not involve third column numbers (e.g 3, 6, 9)

File: tests/test_bin_builder.py [hunk 28]
```diff
@@ -0,0 +1,178 @@
...
+
+        self.bin_builder.build_bins_for_straight_bets(self.wheel)
+
+        for bin_index in straight_outcomes:
+            self.assertIn(straight_outcomes[bin_index], self.wheel.bins[bin_index])
+
+    def test_bins_are_filled_for_horizontal_split_bets(self):
+        split_bet_odds = 17
+        split_outcome_name = "1-2"
+        bin_index = 1
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:28]: if bin 2 is also expected to hold the outcome, better test that too

File: src/roulette.py [hunk 29]
```diff
@@ -126,3 +126,380 @@ def get(self, bin: int) -> Bin:
...
+        a corner bet outcome for each set, each with an associated odds value of 8. Each outcome
+        corresponds to a bet on the intersection of lines covering four adjacent numbers.
+
+        :param wheel: The Wheel object to which the corner bet outcomes will be added.
+        :type wheel: :class:`Wheel`
+
+        The function iterates through sets of four numbers forming a square on the wheel and creates
+        a corner bet outcome for each set,each with an associated odds value of 8. Each outcome
+        corresponds to a bet on the intersection of lines covering four adjacent numbers.
+
```
Comment (FarsanRashid) [inline src/roulette.py:287]: duplicate docstring

File: src/roulette.py [hunk 30]
```diff
@@ -126,3 +126,380 @@ def get(self, bin: int) -> Bin:
...
+        odd_bet = Outcome(odd_bet_name, even_bet_odds)
+
+        for number in even_bet_numbers:
+            if number < 19:
+                wheel.addOutcome(number, low_bet)
+            if number >= 19:
+                wheel.addOutcome(number, high_bet)
+            if number % 2 == 0:
+                wheel.addOutcome(number, even_bet)
+            if number % 2 != 0:
```
Comment (FarsanRashid) [inline src/roulette.py:440]: just use an `else`

File: src/roulette.py [hunk 31]
```diff
@@ -126,3 +126,380 @@ def get(self, bin: int) -> Bin:
...
+                wheel.addOutcome(number, odd_bet)
+            if number in red_bet_numbers:
+                wheel.addOutcome(number, red_bet)
+            if number not in red_bet_numbers:
```
Comment (FarsanRashid) [inline src/roulette.py:444]: can be replaced with an 'else'. that would be more optimized as can avoid the lookup

File: tests/test_bin_builder.py [hunk 32]
```diff
@@ -0,0 +1,178 @@
...
+        red_bet_numbers = {1, 3, 5, 7, 9, 12, 14, 16, 18, 19, 21, 23, 25, 27, 30, 32, 34, 36}
+
+        even_bet_bin_indexes = {1, 17, 18, 36}
+
+        self.bin_builder.build_bins_for_even_money_bets(self.wheel)
+
+        for bin_index in even_bet_bin_indexes:
+            if bin_index in red_bet_numbers:
+                self.assertIn(red_bet_outcome, self.wheel.bins[bin_index])
+            if bin_index not in red_bet_numbers:
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:146]: using `else` is better

File: tests/test_bin_builder.py [hunk 33]
```diff
@@ -0,0 +1,178 @@
...
+                self.assertIn(black_bet_outcome, self.wheel.bins[bin_index])
+            if bin_index < 19:
+                self.assertIn(low_bet_outcome, self.wheel.bins[bin_index])
+            if bin_index > 18:
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:150]: using `else` is better

File: tests/test_bin_builder.py [hunk 34]
```diff
@@ -0,0 +1,178 @@
...
+                self.assertIn(high_bet_outcome, self.wheel.bins[bin_index])
+            if bin_index % 2 == 0:
+                self.assertIn(even_bet_outcome, self.wheel.bins[bin_index])
+            if bin_index % 2 != 0:
```
Comment (FarsanRashid) [inline tests/test_bin_builder.py:154]: using `else` is better

File: tests/test_bin_builder.py (hunk 26 above)
Comment (FarsanRashid) [reply tests/test_bin_builder.py:18]: the existing test is fine. just wanted to share better test strategy.

File: tests/test_bin_builder.py (hunk 34 above)
Comment (FarzanRashid) [reply tests/test_bin_builder.py]: Done.

File: tests/test_bin_builder.py (hunk 33 above)
Comment (FarzanRashid) [reply tests/test_bin_builder.py]: Done.

File: tests/test_bin_builder.py (hunk 32 above)
Comment (FarzanRashid) [reply tests/test_bin_builder.py]: Done.

File: src/roulette.py (hunk 31 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done.

File: src/roulette.py (hunk 30 above)
Comment (FarzanRashid) [reply src/roulette.py]: Done.

File: tests/test_bin_builder.py (hunk 28 above)
Comment (FarzanRashid) [reply tests/test_bin_builder.py]: Done.

File: src/roulette.py (hunk 27 above)
Comment (FarzanRashid) [reply src/roulette.py]: Fixed.

File: tests/test_bin_builder.py (hunk 25 above)
Comment (FarzanRashid) [reply tests/test_bin_builder.py]: Fixed.

File: tests/test_bin_builder.py (hunk 24 above)
Comment (FarzanRashid) [reply tests/test_bin_builder.py]: Done.

File: tests/test_bin_builder.py (hunk 23 above)
Comment (FarzanRashid) [reply tests/test_bin_builder.py]: Done.

File: src/roulette.py (hunk 27 above)
Comment (FarsanRashid) [reply src/roulette.py]: better update the docstring for modified code. also seems black can reformat the updated code @FarzanRashid

File: tests/test_bin_builder.py (hunk 22 above)
Comment (FarsanRashid) [reply tests/test_bin_builder.py:182]: can be done like [this](https://gist.github.com/FarsanRashid/be3d85385376ae72adc45ed01c0797b5)
